lib/msf/core/constants.rb

```diff
@@ -49,6 +49,49 @@ module Msf
     ExcellentRanking => "excellent"
   }
 
+#
+# Stability traits
+#
+
+# Module does not crash the service
+CRASH_SAFE             = 'crash-safe'
+# Module crashes service, but service restarts.
+CRASH_SERVICE_RESTARTS = 'crash-service-restarts'
+# Module crashes service, and service remains down.
+CRASH_SERVICE_DEAD     = 'crash-service-dead'
+# Module crashes the OS, but OS restarts.
+CRASH_OS_RESTARTS      = 'crash-os-restarts'
+# Module crashes the OS, and OS remains down.
+CRASH_OS_DEAD          = 'crash-os-dead'
+# Module causes a resource (such as a file or data in database) to be unavailable for the service.
+SERVICE_RESOURCE_LOSS  = 'service-resource-loss'
+# Modules causes a resource (such as a file) to be unavailable for the OS.
+OS_RESOURCE_LOSS       = 'os-resource-loss'
+
+#
+# Side-effect traits
+#
+
+# Modules leaves payload or a dropper on the target machine
+PAYLOAD_ON_DISK        = 'payload-on-disk'
+# Module modifies some config file on the target machine
+CONFIG_CHANGES         = 'config-changes'
+# Module leaves attack traces in a log file (Example: SQL injection data found in HTTP log)
+MALICIOUS_INPUT_IN_LOG = 'malicious-input-in-log'
```

Comment:
What if my IP is in the log? Yet another artifact that isn't necessarily malicious.

Reply:
Yeah, so the scope I'm looking for isn't all artifacts, it's really the malicious-looking inputs that the user (victim) would be able to identify right away. Like for example, a SQL injection string, you know when you see this, it's bad. In that sense, I wouldn't consider an IP or a user-agent as malicious if they look normal.

Reply:
Interesting specificity. We might want to make sure the evil bit is unset. :-)

Reply:
The constant name doesn't really roll off the tongue, though. Maybe `EXPLOIT_IN_LOGS` or `MALCODE_IN_LOGS` or some such. `INPUT` is kinda specific to your perspective.